splunk-ingestable-s3-bucket/main.tf

```diff
@@ -0,0 +1,203 @@
+// S3 Bucket with a Lifecycle Policy that will delete old items
+resource "aws_s3_bucket" "this" {
+  bucket = "${var.log_bucket_name}"
+  acl    = "private"
+
+  lifecycle_rule {
+    id      = "auto-delete-after-${var.expiration_days}-days"
+    prefix  = ""
+    enabled = true
+
+    expiration {
+      days = "${var.expiration_days}"
+    }
+  }
+
+  tags = "${var.tags}"
+}
+
+// Policy for the bucket with READ and LIST permissions given to the read access account, Other accounts
+// Belonging to the organizaion are only allowed to Put objects to this bucket .
+
+resource "aws_s3_bucket_policy" "this" {
+  bucket = "${aws_s3_bucket.this.id}"
+
+  policy = <<POLICY
+{
+   "Version": "2012-10-17",
+   "Statement": [
+      {
+         "Sid": "splunk_access",
+         "Effect": "Allow",
+         "Principal": {
+            "AWS" : "arn:aws:iam::${var.read_access_account}:root"
+         },
+         "Action": [
+            "s3:Get*",
+            "s3:List*"
+         ],
+         "Resource": [
+            "arn:aws:s3:::${var.log_bucket_name}"
```

Comment:
I might be wrong, but I thought that some S3 operations (e.g. `GetObject`) requires the `/*` - so it GetObject only works for `arn:aws:s3:::somebucket/*` and not `arn:aws:s3:::somebucket`?

Reply:
Yepp, I believe `GetObject` requires `/*`

Reply:
I'll also test if I can omit the List* and Get* permission from the bucket policy altogether. 

The Splunk Integration actually works - with the bucket policy as is. The reason, I think, is that Splunk Assumes a Role in the Account, with an attached policy, including permissions on the bucket.